heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java

```diff
@@ -89,10 +96,15 @@ public Response uploadFile(
     }
 
     String uploadDir = config.getStringValue(FILE_SYSTEM_DIRECTORY);
+    String downloadHostNameOverride = config.getStringValue(DOWNLOAD_HOSTNAME_OVERRIDE);
 
-    String fileName = UUID.randomUUID() + "-" + fileDetail.getFileName();
+    if (StringUtil.isNotBlank(downloadHostNameOverride)) {
```

Comment:
let's add a method that we can use below when forming the uri. I don't like setting a static variable here. 
private String getHostnameOrIp(Config config);
Also you can call config.getString(key, defaultValue) so we don't need the string util.